app/views/incomeSources/add/IncomeSourceCheckDetails.scala.html

```diff
@@ -0,0 +1,208 @@
+        }
+    }
+
+    @accountingMethodMessageKey = @{
+        viewModel.getAccountingMethodMessageKey
+    }
+
+    @changeBusinessNameURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.AddBusinessNameController.changeBusinessNameAgent().url
+        } else {
+            controllers.incomeSources.add.routes.AddBusinessNameController.changeBusinessName().url
+        }
+    }
+
+    @changeBusinessTradeURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.AddBusinessTradeController.show(isAgent = true, isChange = true).url
+        } else {
+            controllers.incomeSources.add.routes.AddBusinessTradeController.show(isAgent = false, isChange = true).url
+        }
+    }
+
+    @changeBusinessAddressURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.AddBusinessAddressController.showAgent(isChange = true).url
+        } else {
+            controllers.incomeSources.add.routes.AddBusinessAddressController.show(isChange = true).url
+        }
+    }
+
+    @getLongAddressFromBusinessAddressDetails(address1: String, address2: Option[String], address3: Option[String], address4: Option[String], postCode: Option[String], countryCode: Option[String]) = @{
+        List(Some(address1),
+            address2,
+            address3,
+            address4,
+            postCode,
+            countryCode
+        ).flatten.filter(_.nonEmpty).mkString("<br>")
+    }
+
+
+    @changeStartDateURL = @{
+        controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.show(isAgent = isAgent, isChange = true, viewModel.incomeSourceType).url
+    }
+
+    @changeAccountingMethodURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.changeIncomeSourcesAccountingMethodAgent(viewModel.incomeSourceType).url
+        } else {
+            controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.changeIncomeSourcesAccountingMethod(viewModel.incomeSourceType).url
+        }
+    }
+
+@mainTemplate(
+    pageTitle = getMessage("title"),
+    backUrl = Some(backUrl),
+    useFallbackBackLink = true,
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @hmrcPageHeading(PageHeading(
+        text = getMessage("title"),
+        headingClasses = Some("govuk-heading-l"),
+        section = Some(getMessage("heading")),
+        captionClasses = Some("govuk-caption-l")
+    ))
+
+    @formWithCSRF(action = postAction) {
+
+        <dl class="govuk-summary-list" id="check-details-table">
+            @if(viewModel.incomeSourceType == SelfEmployment){
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">@getMessage("business-name")</dt>
+                    <dd class="govuk-summary-list__value" id="business-name-value">@{
+                        viewModel.businessName.getOrElse("")
```

Comment:
if there is default value so this would be easy to do this when you constructing model and change field type to be not Optional